package/gluon-mesh-batman-adv/luasrc/lib/gluon/upgrade/330-gluon-mesh-batman-adv-mac-addresses

```diff
@@ -7,9 +7,9 @@ local uci = require('simple-uci').cursor()
 
 -- fix up potentially duplicate MAC addresses (for meshing)
 if not site.mesh.vxlan(true) then
-	uci:set('network', 'wan', 'macaddr', util.generate_mac(0))
+	uci:set('network', 'wan', 'macaddr', util.generate_mac('wan'))
 end
 if uci:get('network', 'mesh_other') then
-	uci:set('network', 'mesh_other', 'macaddr', util.generate_mac(4))
+	uci:set('network', 'mesh_other', 'macaddr', util.generate_mac('mesh_other'))
```

Comment:
I wonder if setting the MAC address here is even necessary... as far as I can tell, it's always set in `210-interface-mesh`?

Reply:
No, the setting of the `wan` MAC address is batman-adv-specific and not done in the generic code.

Reply:
I just tested a mesh_other interface and found that the previous behavior is required.
Without manually setting the macaddr, the mac of `br-mesh_other` was `d8:84:66:8c:9c:60`.
With setting it to `util.generate_mac('mesh_other')` in `network.mesh_other.macaddr` it is correctly: `9a:5b:5b:3c:9c:5c`.

The reason for this is, that the `json_add_string macaddr` is currently only set in `if [ "$vxlan" -eq 1 ]; then`.

FFAC currently does not have VXLAN set for mesh, so this is not executed.
Should I just readd this line or do you think there is some other workaround? @neocturne

Reply:
If there is only one `mesh_interfaces_other` (e.g. br-mesh_other is not created), there is no iface with the mesh_other mac address at all (except for clien1 of course, which has the same offset).

So I wonder why the correct MAC of br-mesh_other is important, while it is not important when having only one port..?

Reply:
Oh, I appear to have misread the code... yea, it needs to be readded, as one piece of code sets the VXLAN interface address, and one the address of the underlying bridge.

Generally we just want to avoid random (or nondeterministically chosen) MAC addresses being visible anywhere. So if there is no bridge, there already is only a single interface with a fixed address. With the bridge, we want to set an address.